// @framer/hyfive.feather-ui-fundamentals/code/Device/iPhone_X_Navigation_Bar (1).tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

// Define type of property
interface Props {
  barColor: string;
  barBgColor: boolean;
}

export class iPhone_X_Navigation_Bar extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    barColor: "dark",
    width: 375,
    height: 34,
    barBgColor: "rgba(255,255,255,1)"
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    barColor: {
      type: ControlType.SegmentedEnum,
      options: ["dark", "light"],
      optionTitles: ["Dark", "Light"],
      title: "Color"
    },
    transparent: {
      type: ControlType.Boolean,
      title: "Transparent"
    },
    barBgColor:{type: ControlType.Color, title:"Bar Color"}
  };

  render() {
    const styles = {
      style: {
        height: "100%",
        display: "flex",
        // alignItems: "center",
        justifyContent: "center",
        textAlign: "center",
        background: this.props.barBgColor,
      },
      dark: {
        width: 134,
        height: 5,
        marginTop: 21,
        borderRadius: 5,
        background: "rgba(0,0,0,.90)"
      },

      light: {
        width: 134,
        height: 5,
        marginTop: 21,
        borderRadius: 5,
        background: "rgba(255,255,255,.90)"
      }
    };
    var styleColor;
    styleColor = styles[this.props.barColor];
    return (
      <div style={styles.style}>
        <div style={styleColor} />
      </div>
    );
  }
}
