// @framer/hyfive.feather-ui-fundamentals/code/Navigation/TitleBar.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";


// Define type of property
interface Props {
  text: string;
  display: string;
  color: string;
  barColor: string;

}

export class TitleBar extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    display: "bar",
    text: "Title",
    color: "rgba(0,0,0,.87)",
    barColor: "rgba(255,255,255,1)",
    width: 375,
    height: 44
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    display: { type: ControlType.Enum, 
        options: ["bar","button"],
        optionTitles: ["Bar","Button"],
        title: "Display" },
    text: { type: ControlType.String, title: "Text" },
    color:{ type: ControlType.Color, title: "Color" },
    barColor:{ type: ControlType.Color, title: "Bar Color" }

  };

  render() {

    const styles = {
        style:{
            height: "100%",
            display: "flex",
            alignItems: "flex-end",
            paddingLeft: 14,
            paddingRight: 14,
            textAlign: "center",
            fontSize: 18,
            background: this.props.barColor,
            overflow: "visible"
        },
        iconBack: {
          position: "absolute",
          bottom: 6,
          left: 18, 
        },
        barBg: {
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
          width: "100%",
          height: 44,
          color: this.props.color,
          // background: "green",
        },
      
        barBgTransparent: {
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
          width: "100%",
          height: 44,
          // background: "green",
        },
      
        circleBtn:{
          position: "absolute",
          width: 40,height: 40,
          background: "white",
          boxShadow: "0px 2px 8px rgba(0,0,0,.1)",
          left: 10,
          borderRadius: 22,
        },
      
        circleBtnTransparent:{
          position: "absolute",
          width: 44,height: 44,
          left: 8,
          borderRadius: 22,
        },
      
        titleTransparent:{
            display: "none",
        }
      
      };


    var barBg, circleBtn, title;

    if(this.props.display=="bar"){
        barBg = styles.barBg;
        circleBtn = styles.circleBtnTransparent


    } else {
        barBg = styles.barBgTransparent;
        circleBtn = styles.circleBtn;
        title = styles.titleTransparent;
        
    }

    return (
      <div style={styles.style}>
        <div style={barBg}>
        <div style={circleBtn}></div>
          <div style={styles.iconBack}>
            <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24">
              <path
                d="M 8.061 12 L 15.53 19.47 C 16.237 20.177 15.177 21.237 14.47 20.53 L 6.47 12.53 C 6.177 12.237 6.177 11.763 6.47 11.47 L 14.47 3.47 C 14.763 3.177 15.237 3.177 15.53 3.47 C 15.823 3.763 15.823 4.237 15.53 4.53 Z"
                fill= {this.props.color}
              />
            </svg>
          </div>
          <div style={title}>{this.props.text}</div>
          
        </div>
      </div>
    );
  }

}
