// @framer/hyfive.feather-ui-fundamentals/code/Form/InputLocation.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

const style: React.CSSProperties = {
    height: "100%",
    display: "flex",
    color: "#8855FF",
    overflow: "hidden",
    background: "white",
};

// Define type of property
interface Props {
    text: string;
    shorter: boolean;
}

export class InputLocation extends React.Component<Props> {

    // Set default properties
    static defaultProps = {
    width: 375,
    height: 116,
    shorter: false,
    }

    // Items shown in property panel
    static propertyControls: PropertyControls = {
    shorter: {type: ControlType.Boolean, title: "Shorter"},
    }

    render() {

    var mapWrapper;
    if(this.props.shorter==true){
        mapWrapper = styles.mapWrapperShorter;
    } else {
        mapWrapper = styles.mapWrapper; 
    }

    return <div style={style}>
    <div style={mapWrapper}><img style={styles.map} src="https://github.com/hyfive/alink.github.io/blob/master/map.png?raw=true"
alt="map"/></div>
    </div>;
    }
}

const styles = {

    mapWrapper:{
        width: "100%", height: 100,
        margin: "12px 20px 4px",
        borderRadius: 8,
        overflow: "hidden",
    },

    mapWrapperShorter:{
        width: "100%", height: 100,
        margin: "12px 20px 4px 60px",
        borderRadius: 8,
        overflow: "hidden",
    },

    map:{
        height: "auto",
        width: "100% ",
    }
}
