// @framer/hyfive.feather-ui-fundamentals/code/Form/InputImage (1).tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

const style: React.CSSProperties = {
  height: "100%",
  display: "flex",
  color: "#8855FF",
  background: "white",
  overflow: "hidden",
  padding: "0 20px"
};

// Define type of property
interface Props {
  text: string;
  sample: string;
  shorter: boolean;
}

export class InputImage extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    sample: "",
    width: 375,
    height: 76,
    shorter: false
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    text: { type: ControlType.String, title: "Text" },
    sample: { type: ControlType.Image, title: "Sample Photo" },
    shorter: { type: ControlType.Boolean, title: "Shorter" }
  };

  svg = (
    <svg xmlns="http://www.w3.org/2000/svg" width="26" height="26">
      <path
        d="M 12.2 12.2 L 0.2 12.2 L 0.2 13.8 L 12.2 13.8 L 12.2 25.8 L 13.8 25.8 L 13.8 13.8 L 25.8 13.8 L 25.8 12.2 L 13.8 12.2 L 13.8 0.2 L 12.2 0.2 Z"
        fill="rgba(0, 0, 0, 0.54)"
      />
    </svg>
  );

  render() {
    const styles = {
      iconAddPhoto: {
        display: "flex",
        margin: "12px 10px 4px 0px",
        width: 60,
        height: 60,
        borderRadius: "var(--radius-m)",
        justifyContent: "center",
        alignItems: "center",
        background: "#F6F7F9"
      },

      iconAddPhotoWithIcon: {
        margin: "12px 12px 4px 60px"
      },

      samplePhoto: {
        width: 60,
        height: 60,
        marginTop: 12,
        marginRight: 12,
        borderRadius: 8,
        overflow: "hidden"
      },

      sample: {
        height: "auto",
        width: "100%"
      },
      noSample: {
        display: "none"
      }
    };

    var samplePhoto, iconAddPhoto;
    if (this.props.sample) {
      samplePhoto = styles.samplePhoto;
    } else {
      samplePhoto = styles.noSample;
    }

    if (this.props.shorter == true) {
      iconAddPhoto = styles.iconAddPhotoWithIcon;
    } else {
      iconAddPhoto = styles.iconAddPhoto;
    }

    return (
      <div style={style}>
        <div style={samplePhoto}>
          <img style={styles.sample} src={this.props.sample} alt="Sample" />
        </div>
        <div style={iconAddPhoto}>{this.svg}</div>
      </div>
    );
  }
}
