// @framer/hyfive.feather-ui-fundamentals/code/Entry/Buttons.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";



// Define type of property
interface Props {
    textL: string;
    textR: string;
    styleL: string;
    styleR: string;
}

export class Buttons extends React.Component<Props> {

    // Set default properties
    static defaultProps = {
    textL: "Left",
    textR: "Right",
    styleL:"outlinedGray",
    styleR:"normal",
    width: 375,
    height: 64,
    }

    // Items shown in property panel
    static propertyControls: PropertyControls = {
    textL: { type: ControlType.String, title: "Text L" },
    textR: { type: ControlType.String, title: "Text R" },

    styleL: {
        type: ControlType.Enum,
        options: ["normal","aciton","pressed","disabled","outlined","outlinedGray","red"],
        optionTitles: ["Normal","action","Pressed","Disabled","Outlined","Outlined-Gray","Normal-Red"],
        title: "Style L",
     },

     styleR: {
        type: ControlType.Enum,
        options: ["normal","action","pressed","disabled","outlined","outlinedGray","red"],
        optionTitles: ["Normal","action","Pressed","Disabled","Outlined","Outline-Gray","Normal-Red"],
        title: "Style R",
     },


    }

    render() {
    
    var styleL;
    styleL = styles[this.props.styleL];

    var styleR;    
    styleR = styles[this.props.styleR];

    return <div style={styles.wrapper}>
        <div style={styleL}>{this.props.textL}</div>
        <div style={styles.gap}></div>
        <div style={styleR}>{this.props.textR}</div>
        </div>;
    }
}


const styles = {

    normal:{
        display: "flex",
        borderRadius: 100,
        height: 44,
        textAlign: "center",
        justifyContent: "center",
        alignItems: "center",
        fontSize: 18,
        fontWeight: 600,
        flex: 1,


        color: "white",
        backgroundColor: "#547EFF",
        
    },

    action:{
        display: "flex",
        borderRadius: 100,
        height: 44,
        textAlign: "center",
        justifyContent: "center",
        alignItems: "center",
        fontSize: 18,
        fontWeight: 600,
        flex: 1,


        color: "white",
        backgroundColor: "#65708A",
        
    },


    pressed:{
        display: "flex",
        borderRadius: 100,
        height: 44,
        textAlign: "center",
        justifyContent: "center",
        alignItems: "center",
        fontSize: 18,
        fontWeight: 600,
        flex: 1,


        color: "white",
        backgroundColor: "#3E80F2",

    },

    disabled:{
        display: "flex",
        borderRadius: 100,
        height: 44,
        textAlign: "center",
        justifyContent: "center",
        alignItems: "center",
        fontSize: 18,
        fontWeight: 600,
        flex: 1,


        color: "white",
        backgroundColor: "#B4CFFF",

    },
    
    outlined:{
        display: "flex",
        borderRadius: 100,
        height: 44,
        textAlign: "center",
        justifyContent: "center",
        alignItems: "center",
        fontSize: 18,
        fontWeight: 600,
        flex: 1,


        color: "#547EFF",
        border: "1px solid #547EFF",
    },

    outlinedGray:{
        display: "flex",
        borderRadius: 100,
        height: 44,
        textAlign: "center",
        justifyContent: "center",
        alignItems: "center",
        fontSize: 18,
        fontWeight: 600,
        flex: 1,


        color: "rgba(0,0,0,.54)",
        border: "1px solid rgba(0,0,0,.2)",
    },

    red:{
        display: "flex",
        borderRadius: 100,
        height: 44,
        textAlign: "center",
        justifyContent: "center",
        alignItems: "center",
        fontSize: 18,
        fontWeight: 600,
        flex: 1,


        color: "white",
        background: "#FF5736",
    },

    // layout:{
    //     padding: "10px 20px",
    // },

    wrapper:{
        display: "flex",
        padding: "10px 20px",
        background: "white",
    },
    gap:{
        display: "flex",
        width: 10,
    }
}
