// @framer/hyfive.feather-ui-fundamentals/code/Entry/Checkbox.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

const style: React.CSSProperties = {
  height: "100%",
  display: "flex",
  alignItems: "center",
  justifyContent: "center"
};

// Define type of property
interface Props {
  checked: boolean;
  disabled: boolean;
}

export class Checkbox extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    checked: true,
    disabled: false,
    width: 24,
    height: 24
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    checked: { type: ControlType.Boolean, title: "Checked" },
    disabled: { type: ControlType.Boolean, title: "Disabled" }
  };

  render() {
    var checker, checkbox;
    if (this.props.checked == true && this.props.disabled == false) {
      checker = "#FFFFFF";
      checkbox = styles.checkboxChecked;
    } else if (this.props.checked == false && this.props.disabled == false) {
      checker = "transparent";
      checkbox = styles.checkbox;
    } else if (this.props.checked == true && this.props.disabled == true) {
      checker = "#FFFFFF";
      checkbox = styles.checkboxCheckedDisabled;
    } else if (this.props.checked == false && this.props.disabled == true) {
      checker = "transparent";
      checkbox = styles.checkboxDisabled;
    }

    return (
      <div style={style}>
        <div style={checkbox}>
          <svg xmlns="http://www.w3.org/2000/svg" width="22" height="22">
            <path
              d="M 9.406 14.112 L 17.719 5.805 C 18.127 5.398 18.787 5.398 19.195 5.806 C 19.602 6.214 19.602 6.874 19.194 7.282 L 10.143 16.325 C 9.736 16.732 9.076 16.732 8.668 16.325 L 3.806 11.467 C 3.398 11.06 3.398 10.399 3.805 9.991 C 4.213 9.584 4.873 9.583 5.281 9.991 Z"
              fill={checker}
            />
          </svg>
        </div>
      </div>
    );
  }
}

const styles = {
  checkbox: {
    display: "flex",
    width: 22,
    height: 22,
    borderRadius: "var(--radius-s)",
    // background: "var(--color-basefill-gray)",
    background: "var(--color-basefill)",
    border: "2px solid rgba(0,0,0,.12)"
  },

  checkboxChecked: {
    display: "flex",
    width: 22,
    height: 22,
    borderRadius: "var(--radius-s)",
    background: "var(--color-brand)"
  },
  checkboxCheckedDisabled: {
    display: "flex",
    width: 22,
    height: 22,
    borderRadius: "var(--radius-s)",
    background: "var(--color-brand)",
    opacity: 0.4
  },
  checkboxDisabled: {
    display: "flex",
    width: 22,
    height: 22,
    borderRadius: "var(--radius-s)",
    background: "var(--color-basefill-gray)",
    border: "2px solid rgba(0,0,0,.12)"
  }
};
