// @framer/hyfive.feather-ui-fundamentals/code/Entry/RadioItem.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import { Radio } from "./Radio";
import "../styles/shared.css";



// Define type of property
interface Props {
  text: string;
  explainer: string;
  checked: boolean;
  type: string;
  disabled: boolean;
}

export class RadioItem extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    text: "Radio",
    checked: true,
    explainer: "",
    height: 59,
    width: 375,
    type: "list",
    disabled: false
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    type: {
      type: ControlType.Enum,
      options: ["form", "list"],
      optionTitles: ["Form / h56", "List / h59 / h80"],
      title: "Type"
    },
    text: { type: ControlType.String, title: "Text" },
    explainer: { type: ControlType.String, title: "Description" },
    checked: {
      type: ControlType.Boolean,
      title: "Checked"
    },
    disabled: {
      type: ControlType.Boolean,
      title: "Disabled"
    }
  };

  render() {

    const styles = {
      style: {
        alignItems: "center"
      },
    
      listText: {
        display: "inline-flex",
        flexDirection: "column",
        color: (this.props.disabled == false)? "var(--color-primary)" : "var(--color-weak)",
      },
      explainer: {
        fontSize: "var(--text-s)",
        color: (this.props.disabled == false)? "var(--color-secondary)":"var(--color-weak)",
        fontWeight: 200
      }
    };
    

    if (this.props.explainer == "") {
      styleList = styles.style;
    }
    var styleList, drop, list, form;

    list = (
      
      <div className="listItemSharedStyle" style={styleList}>
        <div style={styles.listText}>
          <div>{this.props.text}</div>
          <div style={styles.explainer}>{this.props.explainer}</div>
        </div>
        <Radio checked={this.props.checked} disabled={this.props.disabled} />
      </div>
    );

    form = (
      <div className="formItemSharedStyle">
        <Radio checked={this.props.checked}  disabled={this.props.disabled}/>
        <div style={{ paddingLeft: 20, color: (this.props.disabled == false)? "var(--color-primary)" : "var(--color-weak)", }}>{this.props.text}</div>
      </div>
    );

    return <div>{this.props.type == "list" ? list : form}</div>;
  }
}
