// @framer/hyfive.feather-ui-fundamentals/code/Entry/Radio.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import "../styles/shared.css";

const style: React.CSSProperties = {
  height: "100%",
  display: "flex",
  alignItems: "center",
  justifyContent: "center"
};

// Define type of property
interface Props {
  checked: boolean;
  disabled: boolean;
}

export class Radio extends React.Component<Props> {
  static defaultProps = {
    checked: true,
    disabled: false,
    width: 24,
    height: 24
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    checked: { type: ControlType.Boolean, title: "Checked" },
    disabled: { type: ControlType.Boolean, title: "Disabled" }
  };

  render() {
    const styles = {
      radioBg: {
        background:
          this.props.checked == true && this.props.disabled == false
            ? "var(--color-brand)"
            : this.props.checked == false && this.props.disabled == false
            ? "var(--color-basefill)"
            : this.props.checked == true && this.props.disabled == true
            ? "rgba(84,126,255,.4)"
            : "var(--color-basefill-gray)",
        border:
          this.props.checked == true
            ? "transparent"
            : "2px solid rgba(0,0,0,.12)"
      },
      radio: {
        width: 10,
        height: 10,
        borderRadius: 10,
        background: this.props.checked == true ? "white" : "transparent"
      }
    };
    var radio;

    return (
      <div style={style}>
        <div className="radioBg" style={styles.radioBg}>
          <div style={styles.radio} />
        </div>
      </div>
    );
  }
}
