// @framer/hyfive.feather-ui-fundamentals/code/Entry/Switch.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

const style: React.CSSProperties = {
  height: "100%",
  display: "flex",
  alignItems: "center",
  justifyContent: "center",
  textAlign: "center",
  overflow: "hidden"
};

// Define type of property
interface Props {
  state: string;
  disabled: boolean;
}

export class Switch extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    state: "on",
    width: 50, height: 30,
    disabled: false,
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    state: {
      type: ControlType.SegmentedEnum,
      options: ["on", "off"],
      optionTitles: ["On", "Off"],
      title: "State",
    },
    disabled:{
      type: ControlType.Boolean,
      title: "Disabled",
    }
  };

  render() {
    var switchBg, switchKnob, switchBgSharedStyle;

    switchKnob = {
      display: "inline-flex",
      width: 24,
      height: 24,
      background: (this.props.disabled && this.props.state == "off" )? "#E0E0E0":"var(--color-basefill)",
      borderRadius: "var(--radius-cap)",
      boxShadow: this.props.disabled? "none":"var(--shadow-s)"
    };

    switchBgSharedStyle = {
      display: "flex",
      width: 50,
      height: 30,
      padding: "2px 1px",
      alignItems: "center",
      borderRadius: "var(--radius-cap)"
    };

    if (this.props.state == "off") {
      switchBg = {
        display: "flex",
        width: 50,
        height: 30,
        padding: "3px 2px",
        alignItems: "center",
        borderRadius: "var(--radius-cap)",
        background: "var(--color-basefill-gray)",
        
      };
    } else if (this.props.state == "on") {
      switchBg = {
        display: "flex",
        width: 50,
        height: 30,
        padding: "3px",
        alignItems: "center",
        borderRadius: "var(--radius-cap)",
        background: this.props.disabled? "rgba(84,126,255,.4)":"var(--color-brand)",
        flexDirection: "row-reverse",


      };
    }

    return (
      <div style={style}>
        <div style={switchBg}>
          <div style={switchKnob} />
        </div>
      </div>
    );
  }
}
