// @framer/kelc.wechat-ui-kit/code/news_card_item.tsx
import * as React from "react";
import { Frame, PropertyControls, ControlType } from "framer";
import styled from "styled-components";

const NewsCardItem = styled.div`
    width: 100%;
    height: 84px;
    background: white;
    display: grid;
    grid-template-columns: auto 64px;
    grid-column-gap: 16px;
    align-content: center;
    padding: 0 16px 0 16px;
`

const NewsTitle = styled.div`
    width: 100%;
    height: 84px;
    box-sizing: border-box;
    display: grid;
    border-bottom: ${props => props.hasLine ? "0.5px solid #E6E6E6" : "0px"};
    align-items: center;
    
    h2 {
        display: block;
        font-size: 16px;
        line-height: 1.2;
        max-height: 38px;
        overflow: hidden;;
        margin: 0;
        padding-right: 16px;
        font-weight: normal;
        font-family: ".SFNSText", "SFProText-Regular", "SFUIText-Regular", ".SFUIText", sans-serif;
    }
`

const NewsImage = styled.div`
    width: 64px;
    height: 64px;
    overflow: hidden;
    border-radius: 2px;
    background: #0099FF;
    align-self: center;

    img {
        width: 64px;
        height: 64px;
    }
`

// Define type of property
interface Props {
    title: string;
    image: string;
    line: boolean;
}

export class news_card_item extends React.Component<Props> {

    // Set default properties
    static defaultProps = {
        title: "The News Title",
        image: "",
        line: false;
    }

    // Items shown in property panel
    static propertyControls: PropertyControls = {
        title: { type: ControlType.String, title: "Title" },
        image: { type: ControlType.Image, title: "Image" },
        line: { type: ControlType.Boolean, disabledTitle: "Hide", enabledTitle: "Show", title: "Dividing line" },
    }

    render() {
        return (
            <NewsCardItem>
                <NewsTitle hasLine={this.props.line}>
                    <h2>{this.props.title}</h2>
                </NewsTitle>
                <NewsImage>
                    <img src={this.props.image} />
                </NewsImage>
            </NewsCardItem>
        )
    }
}
