// @framer/kelc.wechat-ui-kit/code/switchToolBar.tsx
import { Data, animate, Override, Animatable } from "framer"
import { Curve } from "framer/types/src/animation/Animators/BezierAnimator"

const springOptions = { tension: 500, friction: 30 }

const bezierCurve = [0.2, 0.8, 0.2, 1] as Curve
const duration = 0.6
const bezierOptions = { curve: bezierCurve, duration }

const toolBar = Data({ 
    menuBottom: Animatable(0),
    inputBottom: Animatable(-90),
    showMenu: true,
})

export const SwitchBtn: Override = () => {
    return {
        onTap() {
            if (toolBar.showMenu) {
                animate.bezier(toolBar.menuBottom, -90, bezierOptions);
                animate.bezier(toolBar.inputBottom, 0, bezierOptions);
                toolBar.showMenu = false;
            } else {
                animate.bezier(toolBar.inputBottom, -90, bezierOptions);
                animate.bezier(toolBar.menuBottom, 0, bezierOptions);
                toolBar.showMenu = true;
            }
        },
    }
}

export const MenuBar: Override = () => {
    return {
        bottom: toolBar.menuBottom
    }
}

export const inputBar: Override = () => {
    return {
        bottom: toolBar.inputBottom
    }
}
