// @framer/kelc.wechat-ui-kit/code/Table_view_cell_home.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import { WeChat_Icons_icon_mute } from "./canvas";
import styled from "styled-components";

const CellItem = styled.div`
    display: block;
    position: relative;
    width: 100%;
    height: 68px;
    overflow: hidden;
    padding-left: 10px;
`

const CellContentGroup = styled.div`
    display: grid;
    grid-template-columns: 50px auto;
    height: 68px;
    border-bottom: solid #D9D9D9 0.5px;
`

const Unread = styled.div`
    display: ${props => props.hasUnread ? "block" : "none"}
    position: absolute;
    width: 10px;
    height: 10px;
    border-radius: 5px;
    background: #F43431;
    left: 53px;
    top: 6px;
`

const CellThumbnail = styled.div`
    margin-top: 9px;
    border-radius: 4px;
    overflow: hidden;
    width: 50px;
    height: 50px;
    background-color: #EFEFF3;
    background-image: url(${props => props.image});
    background-size: 50px 50px;
    background-repeat: no-repeat;
    background-position: center;
`
const CellContent = styled.div`
    display: grid;
    grid-template-rows: 1fr;
`

const CellContentLine = styled.div`
    display: grid;
    grid-template-columns: auto auto;
    justify-content: space-between;

    h2, p, span {
        margin: 0;
        display: block;
        font-family: ".SFNSText", "SFProText-Regular", "SFUIText-Regular", ".SFUIText", sans-serif;
    }

    h2 {
        font-size: 17px;
        color: black;
        text-align: left;
        padding-top: 12px;
        padding-left: 10px;
    }

    p {
        font-size: 14px;
        color: #9B9B9B;
        text-align: left;
        padding-bottom: 13px;
        padding-left: 10px;
    }

    span {
        font-size: 12px;
        color: #9B9B9B;
        text-align: right;
        padding-right: 10px;
        padding-top: 13px;
    }
`

const MuteMark = styled.div`
    display: ${props => props.mute ? "block" : "none"};
    width: 17px;
    height: 17px;
    margin-right: 10px;
`

// Define type of property
interface Props {
    title: string;
    subTitle: string;
    time: string;
    thumbnail: string;
    isMute: boolean;
    hasUnread: boolean;
}

export class Table_view_cell_home extends React.Component<Props> {

    // Set default properties
    static defaultProps = {
        title: "Subscriptions",
        subTitle: "[2]news unread...",
        time: "1:03 AM",
        thumbnail: "",
        isMute: false,
        hasUnread: false
    }

    // Items shown in property panel
    static propertyControls: PropertyControls = {
        title: { type: ControlType.String, title: "Title" },
        subTitle: { type: ControlType.String, title: "SubTitle"},
        time: { type: ControlType.String, title: "Time" },
        thumbnail: { type: ControlType.Image, title: "Thumbnail" },
        isMute: { type: ControlType.Boolean, disabledTitle: "Hide", enabledTitle: "Show", title: "Mute Icon" },
        hasUnread: { type: ControlType.Boolean, disabledTitle: "Hide", enabledTitle: "Show", title: "Unread Mark" },
    }

    render() {
        return (
            <CellItem>
                <CellContentGroup>
                    <CellThumbnail image={this.props.thumbnail}>
                        <Unread hasUnread={this.props.hasUnread}></Unread>
                        {this.props.children}
                    </CellThumbnail>
                    <CellContent>
                        <CellContentLine>
                            <h2>{this.props.title}</h2>
                            <span>{this.props.time}</span>
                        </CellContentLine>
                        <CellContentLine>
                            <p>{this.props.subTitle}</p>
                            <MuteMark mute={this.props.isMute}>
                                <WeChat_Icons_icon_mute />
                            </MuteMark>
                        </CellContentLine>
                    </CellContent>
                </CellContentGroup>
            </CellItem>
        );
    }
}
