// @framer/hyfive.feather-ui-fundamentals/code/Map/Map.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

const style: React.CSSProperties = {
  height: "100%",
  display: "flex",
  alignItems: "center",
  justifyContent: "center",
  textAlign: "center",
  color: "#8855FF",
  background: "rgba(136, 85, 255, 0.1)",
  overflow: "hidden"
};

// Define type of property
interface Props {
  zoom: number;
}

export class Map extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    width: 375,
    height: 667,
    zoom: 17
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    zoom: {
      type: ControlType.Number,
      title: "Zoom",
      min: 3,
      max: 18,
      displayStepper: true
    }
  };

  link = {
    zoom_18:
      "https://github.com/hyfive/alink.github.io/blob/master/zoom/25M-HANGZHOU.png?raw=true",
    zoom_17:
      "https://github.com/hyfive/alink.github.io/blob/master/zoom/50M-HANGZHOU.png?raw=true",
    zoom_16:
      "https://github.com/hyfive/alink.github.io/blob/master/zoom/100M-HANGZHOU.png?raw=true",
    zoom_15:
      "https://github.com/hyfive/alink.github.io/blob/master/zoom/200M-HANGZHOU.png?raw=true",
    zoom_14:
      "https://github.com/hyfive/alink.github.io/blob/master/zoom/500M-CHINA.png?raw=true",
    zoom_13:
      "https://github.com/hyfive/alink.github.io/blob/master/zoom/1KM-CHINA.png?raw=true",
    zoom_12:
      "https://github.com/hyfive/alink.github.io/blob/master/zoom/2KM-HANGZHOU.png?raw=true",
    zoom_11:
      "https://github.com/hyfive/alink.github.io/blob/master/zoom/5KM-HANGZHOU.png?raw=true",
    zoom_10:
      "https://github.com/hyfive/alink.github.io/blob/master/zoom/10KM-HANGZHOU.png?raw=true",
    zoom_9:
      "https://github.com/hyfive/alink.github.io/blob/master/zoom/20KM-CHINA.png?raw=true",
    zoom_8:
      "https://github.com/hyfive/alink.github.io/blob/master/zoom/30KM-CHINA.png?raw=true",
    zoom_7:
      "https://github.com/hyfive/alink.github.io/blob/master/zoom/50KM-CHINA.png?raw=true",
    zoom_6:
      "https://github.com/hyfive/alink.github.io/blob/master/zoom/100KM-CHINA.png?raw=true",
    zoom_5:
      "https://github.com/hyfive/alink.github.io/blob/master/zoom/200KM-CHINA.png?raw=true",
    zoom_4:
      "https://github.com/hyfive/alink.github.io/blob/master/zoom/500KM-CHINA.png?raw=true",
    zoom_3:
      "https://github.com/hyfive/alink.github.io/blob/master/zoom/1000KM-WORLD.png?raw=true"
  };

  render() {
    var selectedMap = this.link["zoom_" + this.props.zoom];

    const map = (
      <img src={selectedMap} alt="Feather UI Map" style={styles.map} />
    );

    return <div style={style}>{map}</div>;
  }
}

const styles = {
  map: {
    width: "100%",
    height: "100%",
    objectFit: "cover"
  }
};
