// @framer/hyfive.feather-ui-fundamentals/code/Module/Alert.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

const style: React.CSSProperties = {
    height: "100%",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    textAlign: "center",
    color: "#121212",
    background: "rgba(0, 0, 0, 0.4)",
    overflow: "hidden",
    fontSize: 14,
    lineHeight: 1.5,
};

// Define type of property
interface Props {
    title: string;
    text: string;
    button: string;
}

export class Alert extends React.Component<Props> {

    // Set default properties
    static defaultProps = {
    title: "A Naughty Title",
    text: "Hello, I'm a Alert Dialog, Please feed Me Some Text :)",
    button: "OK",
    width: 375,
    height: 667,
    }

    // Items shown in property panel
    static propertyControls: PropertyControls = {
    title: { type: ControlType.String, title: "Title" },
    text: { type: ControlType.String, title: "Text" },
    button: { type: ControlType.String, title: "Button" },

    }

    render() {
            
  
    return <div style={style}>
    <div style={styles.alert}>
     <div style={styles.title}>{this.props.title}</div>
     <div style={styles.text}>{this.props.text}</div>
     <div style={styles.button}>{this.props.button}</div>
    </div>
    
    </div>;
    }
}

const styles={
    alert:{
        width: 280,
        background: "white",
        borderRadius: 12,
        boxShadow: "0px 2px 8px rgba(0,0,0,.1)",
    },

    title:{
        fontSize: 16,
        fontWeight: 600,
        paddingTop: 16,
    },

    text:{
        width: "100%",
        padding: "8px 20px 16px",  
    },

    button:{
        padding: "12px 0px",
        fontSize: 16,
        fontWeight: 400,
        borderTop: ".5px solid rgba(0,0,0,.1)",
        color:"#547EFF",
    }

}
