// @framer/hyfive.feather-ui-fundamentals/code/Module/Toast.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

const style: React.CSSProperties = {
    height: "100%",
    display: "flex",
    padding: "0 40px",
    alignItems: "center",
    justifyContent: "center",
    textAlign: "center",
    // color: "#8855FF",
    // background: "rgba(136, 85, 255, 0.1)",
    overflow: "visible",
};

// Define type of property
interface Props {
    text: string;
    
}

export class Toast extends React.Component<Props> {

    // Set default properties
    static defaultProps = {
    text: "Toast!",
    height: 667,
    width: 375,
    }

    // Items shown in property panel
    static propertyControls: PropertyControls = {
    text: { type: ControlType.String, title: "Text" },
    }

    render() {
    return <div style={style}> <div style={styles.toast}>{this.props.text}</div></div>;
    }
}


const styles ={
    toast:{
        background: "rgba(0,0,0,.6)",
        color: "white",
        padding: 16,
        borderRadius: 12,
        wordBreak: "break-all",
        lineHeight: 1.5,
        fontSize: 16,

    }
}
