// @framer/hyfive.feather-ui-fundamentals/code/Display/Divider.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import "../styles/style-index.css";

const style: React.CSSProperties = {
  height: "100%",
  display: "flex",

  overflow: "hidden"
};

// Define type of property
interface Props {
  style: string;
  text: string;
}

export class Divider extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    width: 375,
    height: 20,
    style: "line",
    text: "Explainer goes here"
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    style: {
      type: ControlType.SegmentedEnum,
      options: ["line", "none"],
      optionTitles: ["Line", "None"],
      title: "Style"
    },
    text: {
      type: ControlType.String,
      title: "Text / h40",
      hidden(props) {
        return props.style !== "line";
      }
    }
  };

  render() {
    const styles = {
      wrapper: {
        display: this.props.style!="none"?"flex":"none",
        width: "100%",
        justifyContent: "center",
        alignItems: "center",
        fontSize: "var(--text-s)",
        color: "var(--color-weak)",
        padding: "0 20px"
      },
      text: {
        
        position: "absolute"
      },

      p: {
          display: this.props.text?"inline-flex":"none",
        background: "var(--color-basefill)",
        padding: "2px 8px",
        lineHeight: 1.5
      },
      line: {
        display: "flex",
        width: "100%",
        height: 1,
        background: "var(--color-line"
      }
    };

    const divider = (
      <div style={styles.wrapper}>
        <div style={styles.line} />
        <div style={styles.text}>
          <p style={styles.p}>{this.props.text}</p>
        </div>
      </div>
    );
    return <div style={style}>{divider}</div>;
  }
}
