// @framer/hyfive.feather-ui-fundamentals/code/Display/Header.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import "../styles/shared.css";

const style: React.CSSProperties = {
  height: "100%",
  display: "flex",
  alignItems: "center",
  justifyContent: "center",
  textAlign: "center",
  color: "#8855FF",
  background: "rgba(136, 85, 255, 0.1)",
  overflow: "hidden"
};

// Define type of property
interface Props {
  title: string;
  subtitle: string;
  size: string;
}

export class Header extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    title: "Feather UI",
    subtitle: "",
    size: "h1",
    width: 375,
    height: 62,
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    size: {
      type: ControlType.Enum,
      options: ["h1", "h2", "h3", "h4", "h5"],
      optionTitles: ["H1 / h62 / h86", "H2 / h56 / h77", "H3 / h50 / h71", "H4 / h47 / h68", "H5 / h44 / h62"]
    },
    title: { type: ControlType.String, title: "Title" },
    subtitle: { type: ControlType.String, title: "Subtitle" },
    
  };

  render() {
    var styles = {
      h1: {
        fontSize: "var(--h1)",
        fontWeight: 700,
      },
      h2: {
        fontSize: "var(--h2)"
      },
      h3: {
        fontSize: "var(--h3)"
      },
      h4: {
        fontSize: "var(--h4)"
      },
      h5: {
        fontSize: "var(--h5)"
      },
      subtitle :{
          fontSize: this.props.size=="h1"? "var(--text-m)" : this.props.size == "h5"? "var(--text-xs)" : "var(--text-s)",
          fontWeight: 300,
          color: "var(--color-secondary)",
      }
    };



    var headerSize = styles[this.props.size];

    return (
      <div className="headerSharedStyle" >
        <div style={headerSize}>{this.props.title}</div>
        <div style={styles.subtitle}>{this.props.subtitle}</div>
      </div>
    );
  }
}
