// @framer/maxsteitle.progress-bar/code/ProgressBar.tsx
import * as React from 'react';
import { Frame, Animatable, animate, PropertyControls, ControlType, Color } from 'framer';

interface Props {
	width: number;
	height: number;
	percentage: number;
	animationDuration: number;
	primaryColor: string;
	backgroundColor: string;
}

export class ProgressBar extends React.Component<Props> {
	progressBarWidth = Animatable(0);
	animationWidth = Animatable(0);
	animationOpacity = Animatable(1);

	static defaultProps = {
		width: 200,
		height: 8,
		animationDuration: 10,
		primaryColor: '#0055FF',
		backgroundColor: '#CCDDFF'
	};

	static propertyControls: PropertyControls = {
		animationDuration: {
			type: ControlType.Number,
			title: 'Duration',
			min: 1
		},
		primaryColor: {
			type: ControlType.Color,
			title: 'Progress'
		},
		backgroundColor: {
			type: ControlType.Color,
			title: 'Background'
		}
	};

	state = {
		loading: true
	};

	componentDidMount() {
		this.loadingAnimation();
		this.percentageAnimation();
	}

	loadingAnimation = async () => {
		let currentWidth = (this.props.width / this.props.animationDuration) * 3;
		while (this.state.loading === true) {
			await animate.ease(this.animationWidth, currentWidth, {
				duration: 1.5
			}).finished;
			await animate.linear(this.animationOpacity, 0, { duration: 0.6 }).finished;
			await animate.linear(this.animationWidth, 0, { duration: 0 }).finished;
			await animate.linear(this.animationOpacity, 1, { duration: 0.2 }).finished;
			currentWidth = Math.min(
				this.props.width,
				currentWidth + (this.props.width / this.props.animationDuration) * 4.7
			);
		}
	};

	percentageAnimation = async () => {
		await animate.ease(this.progressBarWidth, this.props.width, {
			duration: this.props.animationDuration
		}).finished;
		this.setState({ loading: false });
	};

	render() {
		return (
			<Frame style={outerBar(this.props.backgroundColor)}>
				{innerBar(
					this.progressBarWidth,
					this.props.primaryColor,
					this.animationWidth,
					this.animationOpacity
				)}
			</Frame>
		);
	}
}

function innerBar(
	barWidth: any,
	barColor: string,
	animationWidth: any,
	animationOpacity: any
) {
	return (
		<Frame
			style={{
				height: '100%',
				background: barColor,
				borderRadius: 16,
				position: 'absolute',
				maxHeight: 16
			}}
			width={barWidth}
			left={0}
			top={0}
			overflow={'hidden'}>
			<Frame
				style={{
					background:
						'linear-gradient(-90deg, rgba(255,255,255,.3), rgba(255,255,255,0))',
					height: '100%',
					borderRadius: 16,
					maxHeight: 16
				}}
				width={animationWidth}
				opacity={animationOpacity}
				left={0}
				top={0}
			/>
		</Frame>
	);
}

function outerBar(color: string): React.CSSProperties {
	return {
		height: '100%',
		width: '100%',
		background: color,
		borderRadius: 16,
		overflow: 'hidden',
		maxHeight: 16
	};
}
