// @framer/hyfive.feather-ui-fundamentals/code/Form/InputRemove.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

const style: React.CSSProperties = {
  height: "100%",
  display: "flex",
  alignItems: "center",
  color: "rgba(0,0,0,.38)",
  background: "white",
  overflow: "hidden",
  padding: "16px 20px",
  fontSize: 16
};

// Define type of property
interface Props {
  text: string;
  explainer: string;
  textR: string;
  statusSvg: string;
  statusText: string;
  statusTextR: string;
}

export class InputRemove extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    text: "",
    explainer: "Please Enter Something",
    textR: "Verify",
    statusText: "textLight",
    statusTextR: "textRLight",
    statusSvg: "svg",

    height: 59,
    width: 375
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    statusSvg: {
      type: ControlType.SegmentedEnum,
      options: ["svg", "svgLight"],
      optionTitles: ["Active", "Disabled"],
      title: "Icon Status"
    },

    text: { type: ControlType.String, title: "Text" },
    explainer: { type: ControlType.String, title: "Explainer" },
    // statusText: {
    //   type: ControlType.SegmentedEnum,
    //   options: ["text", "textLight"],
    //   title: "Text Status"
    // },
    textR: { type: ControlType.String, title: "TextR" },
    statusTextR: {
      type: ControlType.SegmentedEnum,
      options: ["textR", "textRLight", "textRTag"],
      title: "TextR Status"
    }
  };

  render() {
    var svg, text, textR;
    svg = styles[this.props.statusSvg];
    text = styles[this.props.statusText];
    textR = styles[this.props.statusTextR];

    return (
      <div style={style}>
        <div style={svg}>
          {" "}
          <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24">
            <path
              d="M 12 0 C 18.627 0 24 5.373 24 12 C 24 18.627 18.627 24 12 24 C 5.373 24 0 18.627 0 12 C 0 5.373 5.373 0 12 0 Z"
              fill="#FF6546"
            />
            <path
              d="M 18.25 11.5 C 18.664 11.5 19 11.836 19 12.25 C 19 12.664 18.664 13 18.25 13 L 5.75 13 C 5.336 13 5 12.664 5 12.25 C 5 11.836 5.336 11.5 5.75 11.5 Z"
              fill="hsl(0, 0%, 100%)"
            />
          </svg>
        </div>
        <div style={styles.textWrapper} >
          <div style={text}><span>
            {this.props.text ? (
              <span style={{ color: "var(--color-primary)" }}>
                {this.props.text}
              </span>
            ) : (
              <span style={{ color: "var(--color-weak)" }}>
                {this.props.explainer}
              </span>
            )}
          </span></div>
          <div style={textR}>{this.props.textR}</div>
        </div>
      </div>
    );
  }
}

const styles = {
  textWrapper: {
    display: "flex",
    width: "100%",
    paddingLeft: 20,
    alignItems: "center",
    justifyContent: "space-between"
  },
  svg: {
    width: 24,
    height: 24
  },

  svgLight: {
    width: 24,
    height: 24,
    opacity: 0.38
  },

  text: {
    color: "rgba(0,0,0,.87)"
  },

  textLight: {
    color: "rgba(0,0,0,.38)"
  },

  textR: {
    // color: "rgba(0,0,0,.87)",
    color: "#65708A",
  },

  textRLight: {
    color: "rgba(0,0,0,.38)",
  },

  textRTag: {
    display: "inline-flex",
    justifyContent: "center",
    alignItems: "center",
    fontSize: 14,
    height: 28,
    padding: "4px 8px",
    background: "#6BB81F",
    color: "white",
    borderRadius: 4,

  }
};
