// @framer/hyfive.feather-ui-fundamentals/code/Form/InputAdd.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";

const style: React.CSSProperties = {
  height: "100%",
  display: "flex",
  alignItems: "center",
  color: "rgba(0,0,0,.38)",
  background: "white",
  overflow: "hidden",
  padding: "16px 20px",
  fontSize: 16
};

// Define type of property
interface Props {
  text: string;
}

export class InputAdd extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    text: "Add Something...",
    height: 59,
    width: 375
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    text: { type: ControlType.String, title: "Text" }
  };

  render() {
    return (
      <div style={style}>
        <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24">
          <path
            d="M 12 0 C 18.627 0 24 5.373 24 12 C 24 18.627 18.627 24 12 24 C 5.373 24 0 18.627 0 12 C 0 5.373 5.373 0 12 0 Z"
            fill="#6BB81F"
          />
          <path
            d="M 11.25 11.25 L 11.25 5.75 C 11.25 5.336 11.586 5 12 5 C 12.414 5 12.75 5.336 12.75 5.75 L 12.75 11.25 L 18.25 11.25 C 18.664 11.25 19 11.586 19 12 C 19 12.414 18.664 12.75 18.25 12.75 L 12.75 12.75 L 12.75 18.25 C 12.75 18.664 12.414 19 12 19 C 11.586 19 11.25 18.664 11.25 18.25 L 11.25 12.75 L 5.75 12.75 C 5.336 12.75 5 12.414 5 12 C 5 11.586 5.336 11.25 5.75 11.25 Z"
            fill="hsl(0, 0%, 100%)"
          />
        </svg>
        <div style={styles.text}>{this.props.text}</div>
      </div>
    );
  }
}

const styles = {
  text: {
    paddingLeft: 20
  }
};
